composeApp/src/commonMain/kotlin/sp/bvantur/tasky/splash/data/SplashLocalDataSource.kt

```diff
@@ -0,0 +1,8 @@
+package sp.bvantur.tasky.splash.data
+
+import sp.bvantur.tasky.core.data.SecurePersistentStorageProvider
+
+class SplashLocalDataSource(private val secureStorageProvider: SecurePersistentStorageProvider) {
+
+    fun isUserAuthorized(): Boolean = secureStorageProvider.isUserAuthorized()
+}
```

Comment:
What is the beenfit that you gain from this additional layer in your architecture? Clean architecture is fine and all, but I personally think we need to adjust it to our project while keeping the benefits. I think I’ve already shared with you that I’ve seen so many projects where I had to click through 5+ classes and layers just to see what a function call does. That doesn’t make code more readable, but much harder to understand.

Instead, you could just call the `SecurePersistentStorageProvider` functions directly in your Repo, which is totally fine since the repo’s abstraction sits in domain.